doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,534 @@
+wt --profile "Windows Powershell" ; --profile "cmd"
+wt --p "Windows Powershell" ; --p "cmd"
+
+# run "my-commandline.exe with some args" in a new tab
+wt new-tab my-commandline.exe with some args
+wt my-commandline.exe with some args
+
+# run "my-commandline.exe with some args and a ; literal semicolon" in a new
+#  tab, and in another tab, run "another.exe running in a second tab"
+wt my-commandline.exe with some args and a \; literal semicolon ; new-tab another.exe running in a second tab
+
+# Start cmd.exe, then split it vertically (with the first taking 70% of it's
+#  space, and the new pane taking 30%), and run wsl.exe in that pane (user story 13)
+wt cmd.exe ; split-pane -t 0 -v -P 30 wsl.exe
+wt cmd.exe ; split-pane -P 30 wsl.exe
+
+# Create a new window with the default profile, create a vertical split with the
+#  default profile, then create a horizontal split in the second pane and run
+#  "media.exe" (user story 13)
+wt new-tab ; split-pane -v ; split-pane -t 1 -h media.exe
+
+```
+
+## `wt` Syntax
+
+The `wt` commandline is divided into two main sections: "Options", and "Commands":
+
+`wt [options] [command ; ]...`
+
+Options are a list of flags and other parameters that can control the behavior
+of the `wt` commandline as a whole. Commands are a semicolon-delimited list of
+commands and arguments for those commands.
+
+If no command is specified in a `command`, then the command is assumed to be a
+`new-tab` command by default. So, for example, `wt cmd.exe` is interpreted the
+same as `wt new-tab cmd.exe`.
+
+<!--
+### Aside: What should the default command be?
+
+These are notes from my draft intentionally left here to help understand the
+conclusion that new-tab should be the default command.
+
+Should the default command be `new-window` or `new-tab`?
+
+`new-window` makes sense to take params like `--initialPosition`,
+`--initialRows`/`--initialCols`, and _implies_ `new-tab`. However, chained
+commands that want to open in the same window _need_ to specify `new-tab`,
+otherwise they'll all appear in new windows.
+
+If it's `new-tab`, then how do `--initialRows` (etc) work? `new-tab` generally
+_doesn't_ accept those parameters, because it's going to be inheriting the
+parent's window size. Do we just ignore them for subsequent invocations? I
+suppose that makes sense, once the first tab has set those, then the other tabs
+can't really change them.
+
+When dealing with a file full of startup commands, we'll assume all of them are
+intended for the given window. So the first `new-tab` in the file will create
+the window, and all subsequent `new-tab` commands will create tabs in that same
+window.
+ -->
+
+TODO: What should we do with an entirely empty command? My gut says _ignore it_.
+For example, `wt ; ; ; ` should not just open 4 tabs. But then what about `wt`
+by itself? We want that to open a new tab with the default profile. So maybe we
+should allow them? Or allow them on the first command only?

```

Comment:
In the case that someone wanted to open 4 tabs at once, what would be the best way to do that? 

Also, looking at the command `wt ; ; ;`, the _expected_ behavior is that it's basically four `wt new-tab` commands right? Would it cause confusion if someone used four empty commands and only got one tab?

Reply:
I guess the explicit way would be to call `wt new-tab ; new-tab ; new-tab ; new-tab`. I just feel a little uncomfortable that something like `wt <some command> ;; <another command>` magically opens another tab in between the first and last commands.

Reply:
Should an empty one not be the equivalent of a repeat or a default case?

In IPv6, the :: repeat is symbolic of 0s the whole way through like `fe80::1` is a short form for `fe80:0000:0000:0000:0000:0000:0000:0001`.

In our debugger WinDBG, just pressing enter is symbolic of running the previous command again. Like `p` for step, <kbd>Enter</kbd>, then just pressing <kbd>Enter</kbd> until you step to the right thing.

I think one of those two patterns would be acceptable for the empty command.

Reply:
Okay I can dig that. I think that using it as a "default case" is probably easier than "repeat the last one", and then `;;` has the same meaning in every context.

I'll update the spec to reflect this discussion :)